system-addon/content-src/components/ContextMenu/ContextMenu.jsx

```diff
@@ -24,9 +24,23 @@ class ContextMenu extends React.Component {
   componentDidUnmount() {
     window.removeEventListener("click", this.hideContext);
   }
-  onKeyDown(index) {
-    if (index > this.props.tabbableOptionsLength) {
-      this.hideContext();
+  onKeyDown(event, option) {
+    switch (event.key) {
+      case "Tab":
+        if (event.shiftKey) {
+          if (option.first) {
+            this.hideContext();
+          }
+          return;
+        }
+        if (option.last) {
+          this.hideContext();
```

Comment:
Oh oops. shift && first || !shift && last. Some reason looks like there should be a simple binary logic for that..